src/view_search.js

```diff
@@ -629,7 +628,7 @@ class view_search extends view
         var elements = this.get_visible_thumbnails(false);
         for(var element of elements)
         {
-            if(element.dataset.illust_id == null)
+            if(element.classList.contains("next-page-placeholder"))
```

Comment:
Removing it asynchronously after the load finishes might be a bit racey, since load_needed_thumb_data might be called multiple times before the load finishes.  I think I'll try removing next-page-placeholder entirely, and just have it trigger the next page load when the last thumb on a page is loaded.

Reply:
Actually it should be fine. ".remove()" will only remove the element from the parent in the document, any reference in the code won't be set to null. In the worst case, it would just call load two times.

Reply:
It's harder to test the interactions, though.  For example, what happens if a load starts and stalls due to network issues, then while it's in progress the user clicks a search?  Another data source will be loaded and start a fetch.  If the old one finishes later, it might not know that it's not the active data source anymore and remove the placeholder at the wrong time.

I'm testing now with the load placeholder removed entirely (f1f45f482), which is a lot simpler than what I was doing before.  Going to test it for a bit longer before sending it out.  Could use the page visibility API now too, but that's a bigger change...